project-addons/flask_middleware_connector/models/stock.py

```diff
@@ -33,10 +36,23 @@ class StockMove(models.Model):
     @api.multi
     def write(self, vals):
         res = super(StockMove, self).write(vals)
-        if vals.get('state', False) and vals["state"] != "draft":
-            for move in self:
-                session = ConnectorSession(self.env.cr, self.env.uid,
+        for move in self:
+            if vals.get('picking_id', False) or (vals.get('state', False) and move.picking_id):
+                vals_picking = {'state': move.picking_id.state}
+                if not vals_picking['state']:
+                    vals_picking = {'state': vals['state']}
+                session = ConnectorSession(self.env.cr, SUPERUSER_ID,
                                            context=self.env.context)
-                on_stock_move_change.fire(session, 'stock.move',
-                                          move.id)
+                order = self.env['sale.order'].search([('name', '=', move.picking_id.origin),
+                                                       ('state', '!=', '')])
```

Comment:
Falta poner el estado

Reply:
Y no me refiero a que pueda estar cancelada la venta, sino un albarán relacionado con la venta, anterior al que se está editando ahora mismo en este write.

Reply:
Si hago un filtrado en el `for picking in order.picking_ids` y le digo que no me haga un fire de los albaranes con estado cancelado, no me elimina los albaranes de Flask ya que no hace unlink.